src/data/matrix/basic.lean

```diff
@@ -203,6 +203,15 @@ begin
   { simp [h, transpose, diagonal_apply_ne' h] }
 end
 
+@[simp] lemma diagonal_conj_transpose [semiring α] [star_ring α] (v : n → α) :
+  (diagonal v)ᴴ = diagonal (star v) :=
+begin
+  ext i j,
+  by_cases h : i = j,
+  { simp [h, conj_transpose] },
+  { simp [h, conj_transpose, diagonal_apply_ne' h], }
+end
```

Comment:
This will be easier to prove if you add it below `diagonal_map`, as then you can prove it with:
```suggestion
begin
  rw [conj_transpose, diagonal_transpose, diagonal_map (star_zero _)],
  refl,
end
```